searx/limiter.py

```diff
@@ -211,23 +211,33 @@ def pre_request():
 
 
 def is_installed():
+    """Returns ``True`` if limiter is active and a redis DB is available."""
     return _INSTALLED
 
 
 def initialize(app: flask.Flask, settings):
-    """Instal the botlimiter aka limiter"""
+    """Install the limiter"""
     global _INSTALLED  # pylint: disable=global-statement
-    if not settings['server']['limiter'] and not settings['server']['public_instance']:
+
+    if not (settings['server']['limiter'] or settings['server']['public_instance']):
         return
+
     redis_client = redisdb.client()
     if not redis_client:
         logger.error(
             "The limiter requires Redis, please consult the documentation: "
-            + "https://docs.searxng.org/admin/searx.botdetection.html#limiter"
+            "https://docs.searxng.org/admin/searx.limiter.html"
         )
         if settings['server']['public_instance']:
             sys.exit(1)
         return
-    botdetection.init(get_cfg(), redis_client)
-    app.before_request(pre_request)
+
     _INSTALLED = True
+
+    cfg = get_cfg()
+    if settings['server']['public_instance']:
+        # overwrite limiter.toml setting
+        cfg.set('botdetection.ip_limit.link_token', True)
```

Comment:
I was thinking about this either at the top of this function or in settings_default.py:

```python
    cfg = get_cfg()
    if settings['server']['public_instance']:
        # overwrite limiter.toml setting
        cfg.set('botdetection.ip_limit.link_token', True)
        # overwrite settings.yml
        settings['server']['limiter'] = True
```

so b05a155 can be revert, and line 222 can check on `settings['server']['limiter']` and there is no need to think about it anymore.

Reply:
> https://github.com/searxng/searxng/commit/b05a15540e1dc2dfb8e4e25aa537b2a68e713844 can be revert

not yet, see the "or" ..

    get_setting('server.limiter') or get_setting('server.public_instance') 

the point is, that `server.public_instance` does not activate the `server.limiter`

I wanted to prepare a PR for this, but first I needed to merge this PR / to prevent a conflict conflict of the new PR and this PR (the new PR needs to be on top oft this PR here).

Reply:
>the point is, that server.public_instance does not activate the server.limiter

ping @unixfox

Reply:
> This new parameter for the moment activate by default both the bot limiter AND the new advanced bot limiter called link_token. 

https://github.com/searxng/searx-instances/discussions/417

So yes it should activate server.limiter